### PR TITLE
Add an Appointment field for Person and a Schedule Command

### LINKED ISSUE #59: Implement the Appointment object as a new field for Person

### PR DESCRIPTION
**Pull request by Kb-Tay**
Add an Appointment field to Person and implement ScheduleCommand.

Previously, person contact did not have have an appointment field. There was no command to schedule appointment.

Add the Appointment class for appointments made and NullAppointment subclass for empty appointments. Added command to add appointment for a person.

This is done so as to allow user to schedule appointment for his clients.

Resolves #59 #60

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ScheduleCommand.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                && toAdd.equals(toAdd);
+    }
+
+    /**
+     * Returns the {@code Person} with the scheduled Appointment.
+     *
+     * @param personToEdit The Person the appointment is scheduled to.
+     * @return The Person with scheduled appointment.
+     */
+    private Person createPersonWithApt(Person personToEdit) {
```
Comment (sopa301) [inline src/main/java/seedu/address/logic/commands/ScheduleCommand.java:82]: maybe createPersonWithAppointment would be a better method name?

File: src/main/java/seedu/address/model/appointment/Appointment.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        return LocalDateTime.parse(input, DATE_FORMATTER);
+    }
+
+    /**
+     * Returns an Appointment after parsing the string representation of the appointment.
+     *
+     * @param appointment The string representation of the appointment.
+     * @return The Appointment object.
+     */
+    public static Appointment parseAptDescription(String appointment) {
```
Comment (sopa301) [inline src/main/java/seedu/address/model/appointment/Appointment.java:86]: parseAppointmentDescription?

File: src/main/java/seedu/address/model/appointment/Appointment.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    /**
+     * Returns the LocaleDateTime object representing the date and time of the Appointment.
+     *
+     * @param input The input date and time from user.
+     * @return The LocaleDateTime object representing Appointment date.
+     * @throws IllegalArgumentException If input date does not match specified format.
+     */
+    public static LocalDateTime parseAptDate(String input) throws IllegalArgumentException {
```
Comment (sopa301) [inline src/main/java/seedu/address/model/appointment/Appointment.java:76]: parseAppointmentDate?

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java
Comment (sopa301) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:1]: Maybe it would be good to have tests for missing or invalid appointments?

**Review by sopa301 — COMMENTED** [review]
Overall pretty good.

**Review by jylow — COMMENTED** [review]
LGTM. Code is abit long can one more person double check thanks

**Review by AlyssaPng — COMMENTED** [review]
Looks Good:)) Though Error/Exception handling. Can add the ScheduleCommand in the UserGuide.

**Comment by sopa301** [comment]
Perhaps we could add tests for the non Javafx functions and classes?

**Review by sopa301 — APPROVED** [review]
LGTM. I assume tests will be added in the future.

**Pull request merged**